app/lib/services/wallet_service.dart

```diff
@@ -37,7 +37,10 @@ Future<List<PkidWallet>> getPkidWallets() async {
   Map<int, dynamic> dataMap = result.asMap();
   final pkidWallets =
       dataMap.values.map((e) => PkidWallet.fromJson(e)).toList();
-  return pkidWallets;
+  Set<String> walletsSet = {};
+  final filteredWallets =
+      pkidWallets.where((p) => walletsSet.add(p.seed)).toList();
+  return filteredWallets;
```

Comment:
please also make sure that the seed is not repeated when adding the wallet

Reply:
What if the user wants to add existing wallet with a diff name?

Reply:
This is not allowed. There is a validation while importing the wallet, but the daily wallet is added in a different way